src/Microsoft.DotNet.MSBuildSdkResolver/Interop.NETStandard.cs

```diff
@@ -16,6 +17,16 @@ internal static partial class Interop
     {
         internal static readonly bool RunningOnWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
 
+        internal static string realpath (string path)
+        {
+            var buffer = new StringBuilder(4096); // PATH_MAX from linux/limits.h
```

Comment:
Ah yes, IIRC now, the marshaler will free native strings returned by p/invoke after copying to a System.String. The real issue is not leak potential, it's ensuring that the automatic free lines up with the native allocation method. I've been bitten by a mismatch on Windows there before. I think CoTaskMemFree is used assuming CoTaskMem. When bitten, I've always dropped to IntPtr and handled conversion and freeing myself as you've done here. 

I don't know if just returning string will cause free() to be used on all Unixes and all runtimes. 
What you have here has no reliance on that so it's more obviously portable and I'm therefore fine with it.